baremetal/metal3machine_manager.go

```diff
@@ -1650,32 +1650,29 @@ func (m *MachineManager) DissociateM3Metadata(ctx context.Context) error {
 	return deleteObject(ctx, m.client, metal3DataClaim)
 }
 
-// getKubeadmControlPlaneName retrieves the KubeadmControlPlane object corresponding to the CAPI machine.
-func (m *MachineManager) getKubeadmControlPlaneName(_ context.Context) (string, error) {
-	m.Log.Info("Fetching KubeadmControlPlane name")
+// getControlPlaneName retrieves the ControlPlane object corresponding to the CAPI machine.
+func (m *MachineManager) getControlPlaneName(_ context.Context) (string, error) {
+	m.Log.Info("Fetching ControlPlane name")
 	if m.Machine == nil {
 		return "", errors.New("Could not find corresponding machine object")
 	}
 	if m.Machine.ObjectMeta.OwnerReferences == nil {
 		return "", errors.New("Machine owner reference is not populated")
 	}
 	for _, mOwnerRef := range m.Machine.ObjectMeta.OwnerReferences {
-		if mOwnerRef.Kind != "KubeadmControlPlane" {
-			continue
-		}
 		aGV, err := schema.ParseGroupVersion(mOwnerRef.APIVersion)
 		if err != nil {
 			return "", errors.New("Failed to parse the group and version")
 		}
 		if aGV.Group != controlplanev1.GroupVersion.Group {
 			continue
 		}
```

Comment:
Looking into a bit more to this, this check should also block fetching CP name other than [KCP and not using v1beta1 APIs](https://github.com/kubernetes-sigs/cluster-api/blob/main/controlplane/kubeadm/api/v1beta1/groupversion_info.go#L30), since we are referencing `controlplanev1.GroupVersion.Group` while comparing the API version..

Reply:
I was wondering about that - this should match any `controlplane.cluster.x-k8s.io` resource I think - I agree that according to the capm3 support matrix it should probably also check the Version, but perhaps we might consider fixing that as a follow-up since it's not strictly related?

Also I note issue which triggered this is related to the RKE2 provider which is [not yet updated to v1beta1](https://github.com/rancher-sandbox/cluster-api-provider-rke2/issues/169) so it may be good if we can ensure that is aligned before enforcing the version here?

Reply:
> I was wondering about that - this should allow any controlplane.cluster.x-k8s.io resource I think - I agree that according to the capm3 support matrix it should probably also check the Version, but perhaps we might consider fixing that as a follow-up since it's not strictly related?

@hardys ah right, it should be fine still if only group matches and not version 👍🏼 

> Also I note issue which triggered this is related to the RKE2 provider which is https://github.com/rancher-sandbox/cluster-api-provider-rke2/issues/169 so it may be good if we can ensure that is aligned before enforcing the version here?

I was commenting on that issue, perhaps you meant to bump API version in the RKE2 provider itself and not the CAPI version? We can discuss the details on that issue itself perhaps 🙂

Reply:
@furkatgofurov7 yes I was thinking the CAPI bump would impact the provider API version but it sounds like that's not the case, so provided we're only checking the group I think this should work OK